service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+
+    // create a data access token (randomly generated UUID) and insert corresponding record into table with parts:
+    // username
+    // uuid
+    // expiration date (current time + 1 month)
+    @Override
+    public DataAccessToken createDataAccessToken(String username, boolean allowRevocationOfOtherTokens) {
+        if (username == null || username.trim().length() == 0) {
+            throw new IllegalArgumentException("subject cannot be empty");
+        } 
+        if (getNumberOfTokensForUsername(username) == maxNumberOfAccessTokens) {
+            if (allowRevocationOfOtherTokens) {
+                revokeOldestDataAccessTokenForUsername(username);
+            } else {
+                throw new MaxNumberTokensExceededException("User has reached max number of tokens allowed (" + maxNumberOfAccessTokens + "). An existing token must expire or be revoked before another token can be assigned.");
+            }
+        }
+        String uuid = UUID.randomUUID().toString();
+        Calendar calendar = Calendar.getInstance();
+        Date creationDate = calendar.getTime();
+        calendar.add(Calendar.SECOND, datTtlSeconds);
+        Date expirationDate = calendar.getTime();
+       
+        DataAccessToken dataAccessToken = new DataAccessToken(uuid, username, expirationDate, creationDate); 
+        dataAccessTokenRepository.addDataAccessToken(dataAccessToken);
+        return dataAccessToken;
+    }
+
+    // get all user tokens/uuids sorted from oldest to newest
+    @Override
+    public List<DataAccessToken> getAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        return allDataAccessTokens;
+    }
+    
+    // get newest data access token for a given username
+    @Override
+    public DataAccessToken getDataAccessToken(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken newestDataAccessToken = allDataAccessTokens.get(allDataAccessTokens.size() - 1);
+        return newestDataAccessToken;
+    }
+
+    @Override
+    public DataAccessToken getDataAccessTokenInfo(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        if (dataAccessToken == null) {
+            throw new TokenNotFoundException("Specified token " + token + " does not exist");
+        }
+        return dataAccessToken;
+    }
+
+    @Override
+    public void revokeAllDataAccessTokens(String username) {
+        dataAccessTokenRepository.removeAllDataAccessTokensForUsername(username);
+    }
+
+    @Override
+    public void revokeDataAccessToken(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
```

Comment:
test: if it doesn't exist what happens?

Reply:
if it doesn't exist it just returns a "success" response -- so I think we should keep the getDataAccessToken check first and throw and exception if the response is null